addons/website_livechat/static/src/new/core/thread_service_patch.js

```diff
@@ -13,4 +13,16 @@ patch(ThreadService.prototype, "website_livechat", {
             });
         }
     },
+    /**
+     * @param {import("@mail/new/core/thread_model").Thread} thread
+     * @param {import('@mail/new/core/persona_model').Persona} persona
+     */
+    avatarUrl(thread, persona) {
+        if (persona?.type === "visitor") {
+            return persona.partner
+                ? `/mail/channel/${thread.id}/partner/${persona.id}/avatar_128`
+                : DEFAULT_AVATAR;
+        }
```

Comment:
override lacks condition on channel existing, and fallback when it's not the case

Reply:
Also what if the visitor persona has a user or a guest rather than a partner?

Reply:
> Also what if the visitor persona has a user or a guest rather than a partner?

Initially, only partner was supported.

https://github.com/odoo-dev/odoo/blob/master-discuss-refactoring/addons/website_livechat/static/src/new/core/persona_model_patch.js#L9